src/features/defi/components/PairIcons/PairIcons.tsx

```diff
@@ -105,10 +109,35 @@ export const PairIcons = ({
 
   if (!icons?.length) return null
 
+  // Combined mode for exactly 2 icons
+  if (icons.length === 2) {
```

Comment:
Are we sure we want a fix for only this count? multiple icons will have the same issue than 2 icons as it will show a +X round

<img width="70" height="51" alt="image" src="https://github.com/user-attachments/assets/06bf19a0-b3bc-47dc-b725-c3a419e5de36" />
 
 In this case the layout will also be weird

Reply:
iIf above 2 we should just show the number icon count. balancer has always done great designing for these use cases. 

We could consider a fancy toggle where if you hit the numbers icon it expands a nested tag with like however many assets are in the pool like this:
<img width="442" height="48" alt="Screenshot 2025-08-28 at 10 59 23 AM" src="https://github.com/user-attachments/assets/e56c9209-0a72-4944-9fde-16c31e52f586" />

tap to expand basically. 

the downside is, of course, doing this on an already expanding row. So we can ticket that experience for the asset itself, since portals keeps throwing these wild multi asset things at us.